### ISSUE TITLE
Save The Game During Combat

### ISSUE THREAD

**Issue by LeHerosInconnu** [ihhub/fheroes2#1218]
A few years ago, I posted this article for April Fool's Day:
https://heroes2.forumactif.com/t539-complete-remake-of-heroes-2

"Currently working on a full remake of Heroes 2 for modern operating systems and graphics tablets.
Windows, Linux, Android and Darwin will be supported to get started.

Will be developed mainly in C++ and Lua LÖVE for maximum flexibility.

The remake will be the "true copy" of our beloved Heroes 2, but without the bugs and with many innovations.

New features:

- 2 new factions with specific towns.
- 3 new campaigns with at least 10 scenarios each.
- new enhanced map editor.
- powerful new campaign editor with possibility of adding additional functionalities (more on this subject later).
- more than 20 new secondary skills.
- more than 40 new spells.
- more than 200 new artifacts (including new ultimate artifacts).
- more than 50 new heroes with portraits and an easy way to add you own heroes by modding.
- advanced modding support for full game features.
- improvement of the AI with an innovative system of "modules" and possibility to add your own modules with specially developed tools.
- complete options system to play the game the way you want.
- multiplayer supported with a PBEM (play by email) system.
- new possibility to save your games during combat to allow PBEM games.
- new Script Action System (SAS) to save and replay your most epic combats.
- multi language support (english and french for first release).
- and much more...

Expect the first release before Christmas!!!

Stay tunned!!!"

In all of this, one thing I wish I could do is to save the current game during a combat, and be able to reload the save file and come back to that point in the combat.
I like to do some combats in different ways and that would save me from having to start the combat all over again.
It would also be really helpful to check for bugs, etc.

The second thing I would also like to do is to be able to save and replay some combats:

"- new Script Action System (SAS) to save and replay your most epic combats."

The combat is simply saved in a text file, easily readable and small in size, showing all the actions performed during a combat from start to finish, starting armies, casting spells, moving and attacking troops, etc.

When this file is loaded in the game, there is the possibility to choose the execution mode, the first mode reproduces only the saved combat, so it's just a replay of the combat.
The second mode allows to take control at the start of the combat and to play the combat with the starting armies against the AI and also to do the combat by letting the AI play for the two heroes.
It could also be very useful for bug reporting and quick testing of AI behavior during combat.

This replay file can be used with the "Battle Only" module already present in fheroes2 (other comments on this module later).

You can read the discussion on this here:
https://heroes2.forumactif.com/t931-fheroes2-save-the-game-during-combat#8445

**Comment by ihhub** [comment]
I don't think that we should even implement this feature. The amount of work needed to do this is very high while the usefulness of this change is little small. Battles in Heroes 2 are much faster compare to older installments in the series so this possible feature would save only at max few minutes of player time.

On top of this a part of this proposal was covered by introducing instant battles. Also for such a feature a UX mockup is needed. I am requalifying this as a discussion till it reaches a maturity state.
